hyperopt/spark.py

```diff
@@ -519,7 +520,7 @@ def run_task_on_executor(_):
                 # The exceptions captured in run_task_on_executor would be returned in the result_or_e
                 finish_trial_run(result_or_e)
 
-        if _spark_major_minor_version >= (3, 2):
+        if self.trials._spark_pinned_threads_enabled:
             from pyspark import inheritable_thread_target
 
             run_task_thread = inheritable_thread_target(run_task_thread)
```

Comment:
@dbczumar 
Could you change the code here to be:
```
        try:
            # pylint: disable=no-name-in-module,import-outside-toplevel
            from pyspark import inheritable_thread_target
            run_task_thread = inheritable_thread_target(run_task_thread)
        except ImportError:
            pass
```

Because `inheritable_thread_target` was introduced in spark 3.2, so the code here will raise importing error if run on spark < 3.2 but enable pin mode.

Reply:
@WeichenXu123 Thank you for the feedback! I've made this change.